lib/acme/client/self_sign_certificate.rb

```diff
@@ -0,0 +1,40 @@
+class Acme::Client::SelfSignCertificate
+  attr_reader :private_key, :subject_alt_names
+
+  extend Forwardable
+  def_delegators :certificate, :to_pem, :to_der
+
+  def initialize(private_key: generate_private_key, subject_alt_names:)
+    @private_key = private_key
+    @subject_alt_names = subject_alt_names
+  end
+
+  def certificate
+    @certificate ||= begin
+      certificate = OpenSSL::X509::Certificate.new
+      certificate.not_before = Time.now
```

Comment:
Let's back date it for an hour to account for clock drift, that seems to be common practice.